packages/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp

```diff
@@ -158,9 +159,18 @@ YogaLayoutableShadowNode::YogaLayoutableShadowNode(
     updateYogaChildren();
   }
 
+  initialize();
   ensureConsistency();
 }
 
+void YogaLayoutableShadowNode::initialize() {
+  if (yogaNode_.style().display() == yoga::Display::Contents) {
+    ShadowNode::traits_.set(ShadowNodeTraits::ForceFlattenView);
+  } else {
+    ShadowNode::traits_.unset(ShadowNodeTraits::ForceFlattenView);
+  }
```

Comment:
Could we fold this into `updateYogaProps` instead to make lifecycle clearer?

Reply:
Done in https://github.com/facebook/react-native/pull/46584/commits/cf58940b5307b0768ce2789873cd360afe856925